### ISSUE TITLE
Tensorflow team is so disappointing in documenting installation on windows pc

### ISSUE THREAD

**Issue by psyg0501** [tensorflow/tensorflow#36385]
Please go to Stack Overflow for help and support:

https://stackoverflow.com/questions/tagged/tensorflow

If you open a GitHub issue, here is our policy:

1. It must be a bug, a feature request, or a significant problem with documentation (for small docs fixes please send a PR instead).
2. The form below must be filled out.
3. It shouldn't be a TensorBoard issue. Those go [here](https://github.com/tensorflow/tensorboard/issues).

**Here's why we have that policy**: TensorFlow developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**:
- **TensorFlow installed from (source or binary)**:
- **TensorFlow version (use command below)**:
- **Python version**:
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**:
- **GPU model and memory**:
- **Exact command to reproduce**:

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with:

```bash
python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"
```

### Describe the problem
Describe the problem clearly here. Be sure to convey here why it's a bug in TensorFlow or a feature request.

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Comment by psyg0501** [comment]
Countless hours of my time have been wasted in trying to install tensorflow CPU version 2.1.0 on my Windows 10 desktop. I have tried all methods in a virtual environment - pip install and conda install methods, ideas given on TF official page, on SO and all over the net. In the end when I enter into Python 3.7 interpeter and type "import tensorflow as tf" is always says "ImportError: DLL load failed: The specified module could not be found." TF official web page gives a URL to use - I used the one that applies to my case: Windows, Python 3.7, CPU version that I want to install. I provide this URL, installation goes through (as always), and in the end the same error crops up. Can the TF team kindly get in touch with Microsoft and fix this mess? That would save millions hours of wasted time globally as I see that the internet is full of such complaints by people trying to install tensorflow. If TF team claims it works, then I am willing to hand over my PC through Teamviewer or such.
[tf_err_msg.docx](https://github.com/tensorflow/tensorflow/files/4141311/tf_err_msg.docx)

**Comment by pdxjohnny** [comment]
Just passing through and saw this. I'd bet you'd have better luck trying in the windows subsystem for linux, although I'm not sure that will give you GPU access.

**Comment by av8ramit** [comment]
Please fill out the template in the future so we are able to better assist you. Looks like [this comment](https://github.com/tensorflow/tensorflow/issues/36167#issuecomment-577886156) can help and is related? We are updating documentation very soon.

**Comment by bmzhao** [comment]
Hi @psyg0501, 

Did you install the Microsoft Visual C++ Redistributable for Visual Studio 2015, 2017, and 2019?

According to the docs at https://www.tensorflow.org/install/pip,

> Install the Microsoft Visual C++ Redistributable for Visual Studio 2015, 2017, and 2019. Starting with the TensorFlow 2.1.0 version, the msvcp140_1.dll file is required from this package (which may not be provided from older redistributable packages). The redistributable comes with Visual Studio 2019 but can be installed separately:

> Go to the [Microsoft Visual C++ downloads](https://support.microsoft.com/en-us/help/2977003/the-latest-supported-visual-c-downloads),
Scroll down the page to the Visual Studio 2015, 2017 and 2019 section.
Download and install the Microsoft Visual C++ Redistributable for Visual Studio 2015, 2017 and 2019 for your platform.
Make sure [long paths](https://superuser.com/questions/1119883/windows-10-enable-ntfs-long-paths-policy-option-missing) are enabled on Windows.

> Install the [64-bit Python 3 release](https://www.python.org/downloads/windows/) for Windows (select pip as an optional feature).
```

Tensorflow 2.1.0 is compiled using MSVC 2019, which appears to require an additional DLL.

**Comment by psyg0501** [comment]
Thank you very much. I installed Visual C++ and then tf2. It works. I still think this requirement should be mentioned in red uppercase letters on tf page. I did not see it there.
Thanks again

Sent from Yahoo Mail on Android 
 
  On Sun, 2 Feb 2020 at 3:19, Amit Patankar<notifications@github.com> wrote:   
Closed #36385.

**Comment by psyg0501** [comment]
Thank you very much. I installed Visual C++ and then tf2. It works. I still think this requirement should be mentioned in red uppercase letters on tf page. I did not see it there.
Thanks again
